Demo 2558 add integration tests

moonlight-komorebi opened the pull request:
## Summary

[NOTE]: # ( Provide a brief overview of the changes. )

* Adding in UATs that will run on changes to the repo. These 

## Checklist
[NOTE]: # ( Just check the ones applicable to this pull request. )
- [x] Documentation updated
- [x] User Acceptance Tests updated

## Reviewers
[NOTE]: # ( Mention reviewers here! )
@aswanson-nr @gabeobrien

moonlight-komorebi committed:
[DEMO-2558] Add placeholder workflow

* Add placeholder workflow for ease of testing. Will be fleshed out in
subsequent commits.
* This should always pass for now so as to not block any current work.

moonlight-komorebi committed:
[DEMO-2558] First attempt at runnable workflow

* Updated workflow to get it into a functional state.
* Added deploy configs & tests in user_acceptance_tests folder. This
is needed for certain steps in the workflow.

moonlight-komorebi committed:
[DEMO-2558] Fixup stop role

moonlight-komorebi committed:
[DEMO-2558] Consolidate user configs

* Update workflow for accomodate the consolidation of user config. Now,
aws, azure, and gcp are all in one user config. Workflow now writes one
user config, with each cloud provider section handling the specific
files needed.

moonlight-komorebi committed:
[DEMO-2558] Swap to using minitest

* Use minitest for testing instead of adhoc script.
* Use Gemfile to manage dependencies. Update workflow to accomodate
this.

moonlight-komorebi committed:
[DEMO-2558] Add slack notification to workflow

moonlight-komorebi committed:
[DEMO-2558] Test using json output

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/newrelic/demo-javatron?pullRequest=8) <br/>All committers have signed the CLA.

moonlight-komorebi committed:
[DEMO-2558] Change workflow name, remove previous test file

* Changed workflow change to its permanent name.
* Remove previously used, but not unused testing script.

moonlight-komorebi committed:
[DEMO-2558] Add README for UATs

aswanson-nr commented on user_acceptance_tests/tests/script.rb:
```diff
@@ -0,0 +1,63 @@
+require 'json'
```
Did you intend to keep this file? It looks like it might be the old version of `tests/happy.rb`

moonlight-komorebi committed:
[DEMO-2558] Update workflow for ease of copying & consuming

* Isolate the repo specific names & values into a single variable, so
that when we need to use this code elsewhere, only that single variable
value needs to change.

moonlight-komorebi committed:
[DEMO-2558] Cleanup stop role

* Previously, two seemingly independent steps were connected. Now, made
the connection more explicit / intuitive.

moonlight-komorebi committed:
[DEMO-2558] Remove push trigger, comment

* Removed push trigger since we only want this to run for PR events.
* Remove no longer needed comment.

aswanson-nr commented on user_acceptance_tests/README.md:
```diff
@@ -0,0 +1,51 @@
+
+* `deployment configuration, deploy config` - a file containing the state of an environment that will be created (or torn down).
+
+* `scenario` - a deployment configuration meant to exercise some specific story or use case.
```
I would avoid using the term `scenario` for this, the sets of actions that the simulator executes are called `scenarios`. Could we stick with using deployment configuration?

moonlight-komorebi replied:
i dont think this is too much of an issue, since someone who contributes / cares about the UATs is likely not a general user (someone who is using the simulator).

if we want a different word, i would not use deployment configuration since this captures extra context. off the top of my head, maybe story, case, use-case is something i would see as next in line.

additionally, the word scenario is also used in the workflow, would you like to see it changed there as well?

aswanson-nr replied:
No, that makes sense! Let's keep it `scenario`

moonlight-komorebi committed:
[DEMO-2558] Replace 'Deployment tests' with 'User Acceptance Tests'

moonlight-komorebi committed:
[DEMO-2558] Fixup stop role to not fail when grep finds no match

* With the addition of 'wc -l' back to the shell command, return code
should always be 0.
* Adjusted checks for subsequent steps to match the new change.

moonlight-komorebi committed:
[DEMO-2558] Update README

moonlight-komorebi committed:
[DEMO-2558] Add github run id to user config name

* This lets multiple invocations of the workflow run without conflicting
with each other -- the resources will have different names since the run
ids will be different.

aswanson-nr commented:
Can you add a link to the user acceptance test readme from the main one?

moonlight-komorebi committed:
[DEMO-2558] Add link to UATs in main README

aswanson-nr reviewed (approved):
Looks good!
